docs: types, news, lua-plugin

justinmk opened the pull request.

skewb1k commented on runtime/doc/news.txt:
```diff
@@ -80,20 +80,17 @@ EVENTS
 LSP
 
-• `root_markers` in |vim.lsp.Config| can now be ordered by priority.
+• Missing fields in LSP messages are now represented using |vim.NIL| instead of nil.
```
```suggestion
• 'null' values in LSP messages are now represented using |vim.NIL| instead of nil.
```
uh oh, looks like I made a typo here back in #34849

justinmk replied:
good catch. i will also add `Missing fields are still represented as nil.`

justinmk commented on runtime/doc/lua-plugin.txt:
```diff
@@ -250,6 +250,32 @@ Validations could include:
+UI                                                           *lua-plugin-ui*
+
+Some plugins have their own "UI" which they present in a buffer that the
+plugin "owns". And in that buffer typically you will want to provide custom
```
Thinking of cases where a plugin creates a buffer for its own purposes, to display its UI.

MariaSolOs replied:
Yeah the quotes make me think that the plugin doesn't really _own_ the buffer.

justinmk commented on runtime/doc/lua.txt:
```diff
@@ -1727,7 +1727,7 @@ vim.isarray({t})                                               *vim.isarray()*
 
     Parameters: ~
-      • {t}  (`table?`)
+      • {t}  (`any?`)
```
That's intentional. `isarray` (and similar type-checking mechanisms) should work on anything.

MariaSolOs commented on runtime/doc/lua-plugin.txt:
```diff
@@ -250,6 +250,32 @@ Validations could include:
+mappings to a specific action by invoking `vim.lsp.buf.code_action()` with the
+`filter` + `apply` parameters: >lua
+
+    vim.lsp.buf.code_action({
```
@skewb1k those are fair arguments. Feel free to comment on the LSP repo issue, I don't want to derail this docs PR with an unrelated discussion :)

justinmk committed:
docs: types, news, lua-plugin

- mention "lua_ls", not "luals". https://github.com/neovim/neovim/discussions/36182

Co-authored-by: Maria Solano <majosolano99@gmail.com>

MariaSolOs commented on runtime/doc/editing.txt:
```diff
@@ -87,6 +87,9 @@ g CTRL-G		Prints the current position of the cursor in five
 			|alternate-file| name.  An unlisted buffer is created
 			to hold the old name.
+
+			See |nvim_buf_set_name()| to avoid filename escaping.
```
I just realized that we're not consistent with using `filename` versus `file name` in our docs.

MariaSolOs commented on runtime/doc/lua-plugin.txt:
```diff
@@ -250,6 +259,32 @@ Validations could include:
+actions.
+
+Besides creating |<Plug>| mappings, you may want to consider providing actions
+by defining an in-process LSP "server". Offering actions as "code-actions"
```
I mentioned this before but I find the quoting confusing. They make me think that the plugin doesn't really own the buffer or that the in-process LSP server isn't a real server.

justinmk committed:
docs(faq): h3 => h2

justinmk committed:
fix(outline): use 2-space indent instead of 1-space

2 spaces is more visually distinct at very little cost.

justinmk committed:
docs: dev_tools, dev_arch

justinmk commented on runtime/doc/dev_tools.txt:
```diff
@@ -6,15 +6,221 @@
+
+
+==============================================================================
+Reproducible build
```
probably should move this to BUILD.md
